pkg/server/router.go

```diff
@@ -188,30 +181,16 @@ func servingKubeletCert(server *config.Control, keyFile string, runtime *config.
 	})
 }
 
-func clientKubeletCert(server *config.Control, keyFile string, runtime *config.ControlRuntime) http.Handler {
-	var secretClient coreclient.SecretClient
+func clientKubeletCert(server *config.Control, keyFile string, auth nodePassBootstrapper) http.Handler {
 	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
-		if secretClient == nil {
-			if runtime.Core == nil {
-				sendError(errors.New("runtime core not ready"), resp)
-				return
-			}
-			secretClient = runtime.Core.Core().V1().Secret()
-		}
-
 		if req.TLS == nil {
 			resp.WriteHeader(http.StatusNotFound)
 			return
 		}
 
-		nodeName, nodePassword, err := getNodeInfo(req)
-		if err != nil {
-			sendError(err, resp)
-			return
-		}
-
-		if err := nodepassword.Ensure(secretClient, nodeName, nodePassword); err != nil {
-			sendError(err, resp, http.StatusForbidden)
+		nodeName, code, err := auth(req)
+		if err != nil || code != 0 {
```

Comment:
It's a little confusing how in some places we're sending back specific HTTP status codes, and in others we're just sending back 0 to mean no error even though it's not a valid HTTP status code. Lets be consistent and treat http.StatusOK as success.
```suggestion
		if err != nil || code != http.StatusOK {
```

Reply:
The idea is that maybe we need want to know about specific error conditions, but explicitly returning the ok code without an error is not particularly helpful. Maybe it is better to drop the code and always return an unauthorized or forbidden if an error is present

Reply:
Yeah it's unclear to me what happens if there's not an error. Does the handler just implicitly return 200 OK with an empty body?

Reply:
It's unclear what gorilla/mux is doing if you don't give it a status code? 
I am not particularly interested in changing that, or the status codes that are returned. Doesn't seem that changes like that are appropriate here without better justification.

Reply:
Even if we don't want to change the conditional sendError, I don't like alternating between returning `http.Status*` consts, or the magic number 0. We should either return bare ints or status codes across the board.

Reply:
Updated the error handling to be simpler and always return `http.StatusForbidden` on secondary auth error. Going to mark these as resolved but feel free to re-open the conversation or create new ones if needed.